### PR TITLE
[Character] Convert Clear/Delete/Get/Update of Character Item Recast to Repositories

### PR DESCRIPTION
**Pull request by Kinglykrab**
# Notes
- Convert `ClearOldRecastTimestamps`, `DeleteItemRecast`, `GetItemRecastTimeStamp`, `GetItemRecastTimestamps`, and `UpdateItemRecast` to repositories.

### CODE REVIEW

File: common/shareddb.cpp [hunk 1]
```diff
@@ -882,34 +883,47 @@ bool SharedDatabase::GetInventory(uint32 account_id, char *name, EQ::InventoryPr
...
-	return Strings::ToUnsignedInt(row[0]);
+	return l[0].timestamp;
 }
 
 void SharedDatabase::ClearOldRecastTimestamps(uint32 char_id)
 {
-	// This actually isn't strictly live-like. Live your recast timestamps are forever
-	const std::string query =
-	    StringFormat("DELETE FROM character_item_recast WHERE id = %u and timestamp < UNIX_TIMESTAMP()", char_id);
-	QueryDatabase(query);
```
Comment (Akkadius) [inline common/shareddb.cpp:912]: We're not doing the same query in the new code

File: common/shareddb.cpp [hunk 2]
```diff
@@ -882,34 +883,47 @@ bool SharedDatabase::GetInventory(uint32 account_id, char *name, EQ::InventoryPr
...
+		)
+	);
+
+	if (l.empty()) {
 		return 0;
+	}
 
-	auto& row = results.begin();
-	return Strings::ToUnsignedInt(row[0]);
+	return l[0].timestamp;
```
Comment (Akkadius) [inline common/shareddb.cpp:921]: Cant we `return l.empty() ? 0 : l[0].timestamp` ?

File: common/shareddb.cpp (hunk 1 above)
Comment (Kinglykrab) [reply common/shareddb.cpp:912]: Fixed.

File: common/shareddb.cpp (hunk 2 above)
Comment (Kinglykrab) [reply common/shareddb.cpp]: Done.

File: zone/zonedb.cpp [hunk 3]
```diff
@@ -3279,23 +3280,25 @@ void ZoneDatabase::RemoveTempFactions(Client *client) {
...
-		timestamp
-	);
-	QueryDatabase(query);
+	auto e = CharacterItemRecastRepository::NewEntity();
+
+	e.id          = character_id;
+	e.recast_type = recast_type;
+	e.timestamp   = timestamp;
+
+	CharacterItemRecastRepository::ReplaceOne(*this, e);
```
Comment (Akkadius) [inline zone/zonedb.cpp:3289]: We could inline

Comment (Kinglykrab) [reply zone/zonedb.cpp]: Done.

**Review by Akkadius — APPROVED** [review]

**Pull request merged**